COS-11: Turn mockup into a functional skeletal

tariqbroadnax opened the pull request:
Created component for dashboard. Added the icons used by the dashboard. Changed the default page to the dashboard.

RussianPrussian commented on src/app/dashboard/dashboard.component.scss:
```diff
@@ -0,0 +1,32 @@
+
+.mat-icon {
+    width: 50px;
+    height: 50px;
```
@asharms26 @NateHand93 I'm no CSS expert, but I'm pretty sure we want to go for "em" units/ percents, ya? Avoid hard pixel values.

asharms26 commented on src/app/dashboard/dashboard.component.scss:
```diff
@@ -0,0 +1,32 @@
+.mat-sidenav-container {
+    height: 600px;
```
Don't modify the sidenav for height. If the Material styles are imported correctly and the sidenav is implemented correctly, the  sidenav should automatically be the full height of it's container.

asharms26 commented on src/app/dashboard/dashboard.component.scss:
```diff
@@ -0,0 +1,32 @@
+}
+
+.mat-icon {
+    width: 50px;
```
If you don't like the size of the icon, then increase it by percentage (> 100%) or decrease it by percentage (< 100%). Also, if this is intended to be a button, then consider a FAB.

asharms26 commented on src/app/dashboard/dashboard.component.scss:
```diff
@@ -0,0 +1,32 @@
+    padding-bottom: 25px;
+}
+
+/deep/ .mat-tooltip {
```
/deep/ is not preferred. It is deprecated/not supported in different browsers.

asharms26 commented on src/app/dashboard/dashboard.component.scss:
```diff
@@ -0,0 +1,32 @@
+p {
+    text-align: center;
+    font-size: 20px;
+    margin-top: 250px;
```
- Don't provide styling to the paragraph element. Give the paragraph element an ID and provide styling to that id.
- Bootstrap has a text-center class that you can use.
- Also, this element would be better served as a header (h1, h2,...,h6).
- If** you are trying to vertically align this element, take a look at Bootstrap 4's grid vertical alignment classes.

asharms26 commented on src/app/app.module.ts:
```diff
@@ -12,30 +12,44 @@ import { ErrorHandler } from '@angular/core';
 	BrowserAnimationsModule,
     HttpClientModule,
-	MatSnackBarModule,
+  MatSnackBarModule,
```
I always find Material cumbersome in the sense that you have to load so many modules for even the slightest little component. To keep this concise, we may be able to consolidate the material widget modules into a single module to be imported into app module. See if you can create a MatWidgetModule and import all the Material widgets to that. Then import the MatWidgetModule into the app module.

asharms26 commented on src/app/dashboard/dashboard.component.ts:
```diff
@@ -0,0 +1,42 @@
+export class DashboardComponent implements OnInit {
+
+  constructor(iconRegistry: MatIconRegistry, sanitizer: DomSanitizer) { 
+    iconRegistry.addSvgIcon(
```
Using Material's icons, we have something else to maintain (the files themselves). Instead, we should be importing FontAwesome (free) into the app. Read up on FontAwesome and how to add the icons to the DOM. With fontawesome, we wont even need to add icons to the iconRegistry.

Unrelated Notes: 

- Way too many lines in the constructor
- You have duplicate code here. What if you could take the addSvgIcon method and nest it within another method that takes two character strings as arguments (hint)?
- We would likely need these icons in other components. It would have been better to add this code such that it could be use in any component within the app.

tariqbroadnax commented:
Here are some screenshots of what it looks like

![image-1](https://user-images.githubusercontent.com/22602857/43857557-9e91af0e-9b19-11e8-92b2-e63197256097.png)

When you hover over a button, you get a tooltip describing the button's action

![image-2](https://user-images.githubusercontent.com/22602857/43857607-ce4feabc-9b19-11e8-92aa-c4e8dbdcf6c3.png)

The following images shows a little how it scales. Images and text largely stay the same size while containers, particularly the dashboard container, scale with the size of the window.

![image](https://user-images.githubusercontent.com/22602857/43857770-4916c41e-9b1a-11e8-911b-5b34454f803f.png)

asharms26 commented:
Remove the power icon. There is no use for it. Instead, move the logout button down there. 

There is also some padding above and below the sidebar. In Chrome, right-click and Inspect Element. You should be able to use the Chrome Developer tools to see where the padding is coming from. Remove that padding.

tariqbroadnax commented:
Top Left button removed. Sidenav now covers full height. Material tooltips replaced with pointed tooltips from another package.

![image](https://user-images.githubusercontent.com/22602857/43923800-c2bd42bc-9bf0-11e8-8805-967793a8ef67.png)

asharms26 commented:
Almost there. Just change the power button icon to something symbolizing logout, and you should be good from me.

tariqbroadnax commented:
Moved login/logout button and changed its icon.

![image](https://user-images.githubusercontent.com/22602857/43970942-3fdbe760-9c9d-11e8-9c94-e8a8d765cf12.png)
